schoolopy/authentication.py

```diff
@@ -0,0 +1,151 @@
+import random
+import time
+import yaml
+import requests_oauthlib
+from requests_oauthlib.oauth1_session import TokenRequestDenied
+import json
+from oauthlib.common import urldecode
+from urllib import parse
+import requests
+
+class AuthorizationError(Exception):
+    """Called when an attempt is made to initialize a Schoology instance with
+    an unauthorized SchoologyAuth instance"""
+    pass
+
+class SchoologyAuth:
+    def __init__(self, consumer_key, consumer_secret, domain_prefix='www', user_id=None, storage_file_name='example_config.yml'):
+        self.API_ROOT = 'https://api.schoology.com/v1'
+        self.DOMAIN_ROOT = 'https://%s.schoology.com' % domain_prefix
```

Comment:
The domain root doesn't always match this pattern. Prefer specifying the entire domain.